crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -354,6 +467,27 @@ impl CumulocityConverter {
             }
             mqtt_messages.append(&mut messages);
             Ok(mqtt_messages)
+        } else if topic.name.starts_with("te/device/") && topic.name.contains("/a/") {
```

Comment:
Is there anything that prevents devices or services from being named e.g. `a`? If not, then that condition is not sufficient.

Reply:
That's right, nothing stops someone from naming a device or a service as `a`/`m`/`e`. Need extra check there, better we check exactly that field. Now changed it to check the exact index of the `a/m/e` substrings after the split.